google/cloud/forseti/services/scanner/dao.py

```diff
@@ -190,3 +199,34 @@ def map_by_resource(violation_rows):
             v_by_type[v_resource].append(v_data)
 
     return dict(v_by_type)
+
+def _create_violation_hash(violation, inventory_index_id,
+                           algorithm=FALLBACK_HASH_ALGORITHM):
+    """Create a hash of violation data using the requested algorithm.
+
+    Args:
+        violation (dict): A violation.
+        inventory_index_id (str): Id of the inventory index.
+        algorithm (str): The algorithm used to create the hash.
+
+    Returns:
+        str: The resulting hex digest or '' if we can't successfully create
+        a hash.
+    """
+    try:
+        violation_hash = hashlib.new(algorithm)
+    except ValueError as e:
+        LOGGER.error('Unable to use the specified hash algorithm: %s',
+                     algorithm)
+        LOGGER.info('Using the fallback hash algorithm: %s',
+                    FALLBACK_HASH_ALGORITHM)
+        violation_hash = hashlib.new(FALLBACK_HASH_ALGORITHM)
```

Comment:
What happens if this hashing fails?

Reply:
As indicated in the method doc it will return an empty string, which is a safe way to fail in this scenario.